erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -912,4 +913,65 @@ def get_paid_amount(dt, dn, party_type, party, account, due_date):
 			and {dr_or_cr} > 0
 	""".format(dr_or_cr=dr_or_cr), (dt, dn, party_type, party, account, due_date))
 
-	return paid_amount[0][0] if paid_amount else 0
\ No newline at end of file
+	return paid_amount[0][0] if paid_amount else 0
+
+
+@frappe.whitelist
+def set_paid_invoices(transactions):
+	frappe.throw("set INvoices")
+	for item in transactions:
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "outstanding_amount", 0)
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "Status", "Paid")
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "discount_amount", item["amount"])
+	frappe.db.commit()
+
+
+# QUESTION: this code should merged with update_invoice_status in the accounts controller?
+@frappe.whitelist
+def cancel_discounted_invoices(transactions):
+	for item in transactions:
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "outstanding_amount", 0)
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "Status", "Paid")
+		frappe.db.set_value(item["reference_doctype"], item["reference_document"], "discount_amount", item["amount"])
+	frappe.db.commit()
+
+
+@frappe.whitelist()
+def calc_discount(discount_double_check):
```

Comment:
https://github.com/frappe/erpnext/wiki/Coding-Standards#break-functions--methods-more-than-10-lines-of-code

Reply:
@rmehta This 10 lines guideline: both JS and py? The link doesn't specify if the style guide is the same for both or not.

Reply:
Obviously both